pkg/pushsync/pushsync.go

```diff
@@ -130,6 +144,9 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				return fmt.Errorf("chunk store: %w", err)
 			}
 
+			// push the chunk to the second closest peers
+			go ps.pushToPeers(ctx, chunk)
```

Comment:
this means that only one push will happen at any given point.
I would inline `pushToPeers` here instead and launch every individual secondry send in a separate goroutine. This way all sends happen simultaneously and one send does not depend on the success of the other in terms of timing (some sends might fail, slowing down other sends)

Reply:
right, I agree